service/history/workflow/transaction_impl.go

```diff
@@ -472,41 +472,44 @@ func updateWorkflowExecutionWithRetry(
 		op, PersistenceOperationRetryPolicy,
 		common.IsPersistenceTransientError,
 	)
-	switch err.(type) {
-	case nil:
-		if namespaceEntry, err := shard.GetNamespaceRegistry().GetNamespaceByID(
-			namespace.ID(request.UpdateWorkflowMutation.ExecutionInfo.NamespaceId),
-		); err == nil {
-			emitMutationMetrics(
-				shard,
-				namespaceEntry,
-				&resp.UpdateMutableStateStats,
-				resp.NewMutableStateStats,
-			)
-			emitCompletionMetrics(
-				shard,
-				namespaceEntry,
-				mutationToCompletionMetric(&request.UpdateWorkflowMutation),
-				snapshotToCompletionMetric(request.NewWorkflowSnapshot),
-			)
-		}
-		return resp, nil
-	case *persistence.CurrentWorkflowConditionFailedError,
-		*persistence.WorkflowConditionFailedError,
-		*persistence.ConditionFailedError:
-		// TODO get rid of ErrConflict
-		return nil, consts.ErrConflict
-	default:
+	if err != nil {
 		shard.GetLogger().Error(
-			"Persistent store operation Failure",
+			"Update workflow execution operation failed.",
 			tag.WorkflowNamespaceID(request.UpdateWorkflowMutation.ExecutionInfo.NamespaceId),
 			tag.WorkflowID(request.UpdateWorkflowMutation.ExecutionInfo.WorkflowId),
 			tag.WorkflowRunID(request.UpdateWorkflowMutation.ExecutionState.RunId),
 			tag.StoreOperationUpdateWorkflowExecution,
 			tag.Error(err),
 		)
-		return nil, err
+		switch err.(type) {
+		case *persistence.CurrentWorkflowConditionFailedError,
+			*persistence.WorkflowConditionFailedError,
+			*persistence.ConditionFailedError:
+			// TODO get rid of ErrConflict
+			return nil, consts.ErrConflict
+		default:
+			return nil, err
+		}
 	}
```

Comment:
if the intention is to emit error logs whenever error is encountered, then i would suggest log when error is generated

Reply:
I follow the rule: "Error should be propagated or handled. Not both". It gets propagated to this point and this substitution is sort of handling and good place to log it. I know we lost error call stack in this case.